### DISCUSSION TITLE
Uploading file as part of an object

### DISCUSSION THREAD

**Discussion by dandanthedev** [pocketbase/pocketbase#3062] [Q&A]
Hi, im making an edit function for my records in sveltekit but not getting the file changes to work.
i just have a simple `<input type="file"/>` in my html and all the other elements.

this is my update code:
```
let form = document.querySelector('.form');
						let formData = new FormData(form);
						let data = {};
						//set only the changed values
						for (let [key, value] of formData.entries()) {
							if (key == 'situaties') {
								thema[key] = JSON.stringify(thema[key]);
								value = JSON.stringify(JSON.parse(value));
							}
							if ((key == 'boekje' || key == 'img') && value.name == '') continue;

							if (value != thema[key]) {
								if (key == 'boekje' || key == 'img') {
									//make imgs work correctly here
								}

								data[key] = value;
							}
							if (key == 'situaties') thema[key] = JSON.parse(value);
						}
						console.log(data);
						if (Object.keys(data).length == 0) return alert('Geen veranderingen');

						let res = await pb.collection('themas').update(thema.id, data);
						if (typeof res.$export() !== 'undefined') {
							alert('Thema geupdate');
							window.location.reload();
						} else {
							alert('Er is iets fout gegaan');
						}
```
How can i make the file work with pocketbases js sdk?

**Comment by ganigeorgiev** [comment]
You need to use `FormData` when performing file upload so that the `fetch` can send `multipart/form-data` request.

For an example, please check https://pocketbase.io/docs/files-handling/#uploading-files.

_Option to allow submitting `Blob/File` instances as part of a plain js object will be added in the future (see https://github.com/pocketbase/js-sdk/issues/209)._

**Comment by dandanthedev** [comment]
> You need to use `FormData` when performing file upload so that the `fetch` can send `multipart/form-data` request.
> 
> For an example, please check https://pocketbase.io/docs/files-handling/#uploading-files.
> 
> _Option to allow submitting `Blob/File` instances as part of a plain js object will be added in the future (see [pocketbase/js-sdk#209](https://github.com/pocketbase/js-sdk/issues/209))._

How would i go about doing this? As far as i can see im using it correctly.

**Reply by ganigeorgiev** [reply]
> As far as i can see im using it correctly.

No, you are passing and creating a new plain object, aka. the `data` variable, and not a `FormData` instance. You can either convert the `data` object to `FormData` or directly modify the previously created `formData` variable.

For the available FormData methods, please refer to https://developer.mozilla.org/en-US/docs/Web/API/FormData.
